File: 5_wilcoxon_signed_rank_test.py
import pandas as pd
from scipy.stats import wilcoxon

# Read the CSV data into a DataFrame
data = pd.read_csv('your_data.csv')

# Perform the Wilcoxon signed-rank test
result = wilcoxon(data['Population'], data['AverageIncome'])

# Print the test statistic and p-value
print("Wilcoxon signed-rank test:")
print("Test Statistic:", result.statistic)
print("p-value:", result.pvalue)

File: 5_wilcoxon_signed_rank_test.r
# Read the CSV data into a DataFrame
data <- read.csv("your_data.csv")

# Perform the Wilcoxon signed-rank test
result <- wilcox.test(data$Population, data$AverageIncome, paired = TRUE)

# Print the test statistic and p-value
cat("Wilcoxon signed-rank test:\n")
cat("Test Statistic:", result$statistic, "\n")
cat("p-value:", result$p.value, "\n")
